sdk/go/common/util/gitutil/git.go

```diff
@@ -54,8 +54,8 @@ const (
 
 	// GitLabHostName The host name for GitLab.
 	GitLabHostName VCSKind = "gitlab.com"
-	// GitHubHostName The host name for GitHub.
-	GitHubHostName VCSKind = "github.com"
+	// GitHubHostName The host name for GitHub. This is not the full hostname to account for GitHub Enterprise.
+	GitHubHostName VCSKind = "github"
```

Comment:
I don't think this is the right change.  `GitHubHostName` is used in other places as well, e.g. https://github.com/pulumi/pulumi/blob/master/pkg/cmd/pulumi/deployment/deployment_settings_config.go#L433, which will break with this change.  

`GitLabHostName` seems less used, but I think it refers to `gitlab.com` always right now.  (Kinda wondering if we should get rid of these constants completely instead, but that's a separate issue).

Also we should give the same treatment to `GitLab`.